### ISSUE TITLE
Plugins Marketplace: Inconsistent tag name casing

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/wp-calypso#93592]
## Description

In the Plugins Marketplace, the tag name casing in the plugin detail page is consistent. Sometimes it seems like a slug, and sometimes it's capitalized. See for example:

![Image](https://github.com/user-attachments/assets/e037c923-8b19-472d-aa18-ad4e273256d9)

Tag names should be displayed in sentence case.

**Comment by mmtr** [comment]
> Sometimes it seems like a slug, and sometimes it's capitalized

I think this a Core issue as well caused by inconsistent tags added by plugin authors:

- https://wordpress.org/plugins/wordpress-seo/
<img width="365" alt="Screenshot 2024-08-19 at 13 50 15" src="https://github.com/user-attachments/assets/2e65c73b-1b82-4fc6-84ed-d889b83d0c04">

- https://wordpress.org/plugins/woocommerce
<img width="369" alt="Screenshot 2024-08-19 at 13 49 43" src="https://github.com/user-attachments/assets/6c2b70c0-f715-4d09-b106-e11a7584834c">

- https://wordpress.org/plugins/jetpack/
 
<img width="364" alt="Screenshot 2024-08-19 at 13 50 02" src="https://github.com/user-attachments/assets/27d33dfc-8ffe-4eb5-9a57-f60764eb91b3">

> Tag names should be displayed in sentence case.

Are we sure about this? They look better to me in lowercase

**Comment by DustyReagan** [comment]
I jumped the gun a little on this one before seeing the "Needs more details" tag. 😅 I already created a PR of my proposal above. 👆 

I agree with @mmtr for two reasons. Transforming to lowercase is more reliable for consistency since sometimes we get tags like "eCommerce" or "BBQ," which sentence case would transform to "Ecommerce" and "Bbq." Secondly, if you click on a plugin tag, the browse page you land on (/plugins/browse/content-analysis, for example) already transforms the tag to lowercase. So, in this way, we'd match.
